### ISSUE TITLE
support player referrals

### ISSUE THREAD

**Issue by seiyria** [After-the-End-of-All-Things/game#43]
- each player should have a referral id (it can just be their mongo id)
- new field on user schema to track referral
- new field on register to track referrals
- players should have a card that shows how many referrals they have
- players should get referral rewards for any referred player who reached level 10, up to a certain number of referrals (premium currency and coins)
- players who reach level 10 and were referred should get less premium currency, but still some
- when ?ref=X is in the url, it should auto populate referral field on registration, and clear it from the url. it can also store this in local storage.
- send a notification to the referrer and referree when they can claim their free stuff
- support a QR code to share your player profile (cuteify the qr code to have a center icon for the players sprite)

**Comment by DrKain** [comment]
I like the idea of referrals but it's worth mentioning that eventually some player will try cheat the system with bogus/throwaway accounts.  
You could probably limit this by verifying emails, linking accounts to Discord or even checking the referral IP, just food for thought though

**Comment by seiyria** [comment]
Good call, this is probably better served as a way later thing, until I have discord linking sorted out. Thanks!
